index.js

```diff
@@ -0,0 +1,143 @@
+    "import/no-mutable-exports": "error",
+    "import/no-named-as-default-member": "error",
+    "import/no-named-default": "error",
+    "import/no-self-import": "error",
+    "import/no-unresolved": "error",
+    "import/no-useless-path-segments": "error",
+    "import/no-webpack-loader-syntax": "error",
+    "import/order": ["error", { "newlines-between": "always" }],
+    "import/prefer-default-export": "off",
+    "jsx-a11y/anchor-is-valid": "error",
+    "react/boolean-prop-naming": "error",
+    "react/button-has-type": "error",
+    "react/default-props-match-prop-types": "error",
+    "react/destructuring-assignment": [
+      2,
+      "always",
+      { ignoreClassFields: true }
+    ],
+    "react/forbid-dom-props": ["error", { forbid: ["style", "id"] }],
+    "react/forbid-prop-types": "error",
+    "react/jsx-boolean-value": ["error", "never"],
+    "react/jsx-child-element-spacing": "error",
+    "react/jsx-closing-bracket-location": "error",
+    "react/jsx-closing-tag-location": "error",
+    "react/jsx-curly-brace-presence": [
+      "error",
+      { props: "never", children: "never" }
+    ],
+    "react/jsx-curly-spacing": "error",
+    "react/jsx-equals-spacing": "error",
+    "react/jsx-filename-extension": ["error", { extensions: [".js"] }],
+    "react/jsx-first-prop-new-line": "error",
+    "react/jsx-handler-names": "error",
+    "react/jsx-indent": ["error", 2],
+    "react/jsx-key": "error",
+    "react/jsx-max-depth": ["error", { max: 4 }],
+    "react/jsx-no-bind": "error",
+    "react/jsx-no-comment-textnodes": "error",
+    "react/jsx-no-duplicate-props": "error",
+    "react/jsx-no-target-blank": "error",
+    "react/jsx-one-expression-per-line": "error",
+    "react/jsx-pascal-case": "error",
+    "react/jsx-props-no-multi-spaces": "error",
+    "react/jsx-sort-default-props": "error",
+    "react/jsx-tag-spacing": ["error", { beforeClosing: "never" }],
+    "react/jsx-uses-vars": "error",
+    "react/jsx-wrap-multilines": [
+      "error",
+      {
+        declaration: "parens-new-line",
+        assignment: "parens-new-line",
+        return: "parens-new-line",
+        arrow: "parens-new-line",
+        condition: "parens-new-line",
+        logical: "parens-new-line",
+        prop: "parens-new-line"
+      }
+    ],
+    "react/forbid-foreign-prop-types": "error",
+    "react/no-access-state-in-setstate": "error",
+    "react/no-array-index-key": "error",
+    "react/no-children-prop": "error",
+    "react/no-danger": "error",
+    "react/no-deprecated": "error",
+    "react/no-did-mount-set-state": "error",
+    "react/no-did-update-set-state": "error",
+    "react/no-direct-mutation-state": "error",
+    "react/no-find-dom-node": "error",
+    "react/no-is-mounted": "error",
+    "react/no-multi-comp": "error",
+    "react/no-render-return-value": "error",
+    "react/no-typos": "error",
+    "react/no-string-refs": "error",
+    "react/no-this-in-sfc": "error",
```

Comment:
There's an eslint rule to avoid using `this` in anything that's not a class I think. But I don't think we want that. There may be cases where we want to reference the current object we are in, or whatever `this` is in certain contexts.

This is the rule: https://eslint.org/docs/rules/no-invalid-this

Reply:
I think that these cases aren't very common 🤔 In this case, I'd disable the linter rule. In my experience, dealing with `this` in a `mapStateToProps` due to a bad copy/paste was painful, specially because didn't have too much experience with this tech

Reply:
Sorry, I'm confused about which rule we're talking about. Do you like or dislike this rule (react/no-this-in-sfc)? I can't find any harm on having it, you just shouldn't use `this` in stateless functional components.

On the other hand, I think we can discuss having the `no-invalid-this` rule. General `eslint` rules will come from a config repo we'll share with Frontend and Node, and I think we agreed on not having that rule, but we could override it here if we want. I don't have a strong opinion on this, I'm leaning a little bit towards not having the rule because it's not a "bad practise", only something that could be an issue if we don't understand what `this` really is in each context. The solution would be to understand it better more than disabling something that could be useful.

But again, I'm still not sure if avoiding possible bugs given that it's not very common having to use `this` outside a class could be a better tradeoff.

@WFolini @SKOLZ what do you think?

Reply:
I like this rule. I don't see the reason for not implementing it and doing the opposite, using `this` in sfc. My vote is a `yes`

Reply:
@damfinkel I like this rule but was wondering if this can be extensible to other use cases 😇

Reply:
`no-invalid-this` will be set as "error" in `eslint-config-wolox`